core/templates/pages/admin-page/features-tab/admin-features-tab.component.ts

```diff
@@ -0,0 +1,249 @@
+      placeholder?: string;
+      inputRegex?: RegExp;
+    }
+  } = {
+    [PlatformParameterFilterType.ServerMode]: {
+      displayName: 'Server Mode',
+      options: AdminFeaturesTabConstants.ALLOWED_SERVER_MODES,
+      operators: ['='],
+      optionFilter: (feature, option) => {
+        switch (feature.featureStage) {
+          case FeatureStage.DEV:
+            return option === 'dev';
+          case FeatureStage.TEST:
+            return option !== 'prod';
+          case FeatureStage.PROD:
+            return true;
+        }
+      }
+    },
+    [PlatformParameterFilterType.UserLocale]: {
+      displayName: 'User Locale',
+      options: AdminFeaturesTabConstants.ALLOWED_SITE_LANGUAGE_IDS,
+      operators: ['=']
+    },
+    [PlatformParameterFilterType.ClientType]: {
+      displayName: 'Client Type',
+      options: AdminFeaturesTabConstants.ALLOWED_CLIENT_TYPES,
+      operators: ['=']
+    },
+    [PlatformParameterFilterType.BrowserType]: {
+      displayName: 'Browser Type',
+      options: AdminFeaturesTabConstants.ALLOWED_BROWSER_TYPES,
+      operators: ['=']
+    },
+    [PlatformParameterFilterType.AppVersion]: {
+      displayName: 'App Version',
+      operators: ['=', '<', '>', '<=', '>='],
+      placeholder: 'e.g. 1.0.0',
+      inputRegex: AdminFeaturesTabConstants.APP_VERSION_REGEXP
+    },
+    [PlatformParameterFilterType.AppVersionFlavor]: {
+      displayName: 'App Version Flavor',
+      options: AdminFeaturesTabConstants.ALLOWED_APP_VERSION_FLAVORS,
+      operators: ['=', '<', '>', '<=', '>=']
+    }
+
+  };
+
+  featureFlags: PlatformParameter[] = [];
+  featureFlagNameToBackupMap: Map<string, PlatformParameter>;
+
+  constructor(
+    private windowRef: WindowRef,
+    private adminDataService: AdminDataService,
+    private adminTaskManager: AdminTaskManagerService,
+    private apiService: PlatformFeatureAdminBackendApiService,
+    private ruleFactory: PlatformParameterRuleObjectFactory,
+    private filterFactory: PlatformParameterFilterObjectFactory,
+  ) {}
+
+  async reloadFeatureFlagsAsync(): Promise<void> {
+    const data = await this.adminDataService.getDataAsync();
+
+    this.featureFlags = data.featureFlags;
+
+    this.featureFlagNameToBackupMap = new Map(
+      this.featureFlags.map(feature => [feature.name, cloneDeep(feature)]));
+  }
+
+  addNewRuleToTop(feature: PlatformParameter): void {
+    feature.rules.unshift(
+      this.ruleFactory.createFromBackendDict({
+        filters: [{
+          type: PlatformParameterFilterType.ServerMode,
+          conditions: [['=', ServerMode.Dev.toString()]]
+        }],
+        value_when_matched: false
+      })
+    );
+  }
+
+  addNewRuleToBottom(feature: PlatformParameter): void {
+    feature.rules.push(
+      this.ruleFactory.createFromBackendDict({
+        filters: [{
+          type: PlatformParameterFilterType.ServerMode,
+          conditions: [['=', ServerMode.Dev.toString()]]
+        }],
```

Comment:
Maybe save this as a private constant to represent the default filter (and to provide context that's what this is).